testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java

```diff
@@ -193,6 +195,67 @@ public void testCreateSchema()
     }
 
+    @Test
+    public void testRenameTable()
+    {
+        if (!hasBehavior(SUPPORTS_RENAME_TABLE)) {
+            // must reference an existing table in ALTER TABLE since name is validated before calling renameTable
+            assertQueryFails("ALTER TABLE nation RENAME TO yyyy", "This connector does not support renaming tables");
+            return;
+        }
+
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            throw new AssertionError("Cannot test ALTER TABLE RENAME without CREATE TABLE, the test needs to be implemented in a connector-specific way");
+        }
+
+        String oldTable = "test_rename_" + randomTableSuffix();
+        String newTable = "test_rename_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + oldTable + " (a bigint, b double)");
+        assertUpdate("ALTER TABLE " + oldTable + " RENAME TO " + newTable);
+        assertThat(query("SELECT a, b FROM " + newTable))
+                .returnsEmptyResult();
+        assertUpdate("INSERT INTO " + newTable + " (a, b) VALUES (42, -38.5)", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + newTable))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+        assertUpdate("DROP TABLE " + newTable);
+    }
+
+    @Test
+    public void testRenameTableAcrossSchemas()
+    {
+        if (!hasBehavior(SUPPORTS_RENAME_TABLE_ACROSS_SCHEMAS)) {
+            // must reference an existing table in ALTER TABLE since name is validated before calling renameTable
+            assertQueryFails("ALTER TABLE nation RENAME TO other_schema.yyyy", "This connector does not support renaming tables");
+            return;
+        }
+
+        if (!hasBehavior(SUPPORTS_CREATE_SCHEMA)) {
+            throw new AssertionError("Cannot test ALTER TABLE RENAME across schemas without CREATE SCHEMA, the test needs to be implemented in a connector-specific way");
+        }
+
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            throw new AssertionError("Cannot test ALTER TABLE RENAME across schemas without CREATE TABLE, the test needs to be implemented in a connector-specific way");
+        }
+
+        String schemaName = "test_schema_create_" + randomTableSuffix();
+        assertUpdate("CREATE SCHEMA " + schemaName);
+
+        String oldTable = "test_rename_" + randomTableSuffix();
+        String newTable = schemaName + ".test_rename_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + oldTable + " (a bigint, b double)");
+        assertUpdate("ALTER TABLE " + oldTable + " RENAME TO " + newTable);
+        assertThat(query("SELECT a, b FROM " + newTable))
+                .returnsEmptyResult();
+        assertUpdate("INSERT INTO " + newTable + " (a, b) VALUES (42, -38.5)", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + newTable))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+        assertUpdate("UPDATE " + newTable + " SET a = 43, b = -39.5", 1);
```

Comment:
remove UPDATE, INSERT should suffice.

also UPDATE will make this test not very reusable, since it's supported for Hive ORC ACID only, today